wiseconnect/components/device/silabs/si91x/mcu/drivers/peripheral_drivers/src/rsi_rng.c

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+* @file  rsi_rng.c
+* @brief 
+*******************************************************************************
+* # License
+* <b>Copyright 2022 Silicon Laboratories Inc. www.silabs.com</b>
+*******************************************************************************
+*
+* The licensor of this software is Silicon Laboratories Inc. Your use of this
+* software is governed by the terms of Silicon Labs Master Software License
+* Agreement (MSLA) available at
```

Comment:
I think we should make it a high priority task to get rid of MSLA from the downstream HAL tree, i.e. not just fix all existing files but also stop introducing new files with that license.

What exactly is needed to accomplish this?

Note: I don't mean to block this PR, since legal approvals shouldn't hold up technical downstream work, but it's just worrying to see the number of MSLA files growing and growing when we should see zlib gradually starting to take over the entire tree.